nitrocli/src/options.rs

```diff
@@ -0,0 +1,155 @@
+      }
+    )
+  }
+}
+
+impl FromStr for Command {
+  type Err = ();
+
+  fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+    match s {
+      "clear" => Ok(Command::Clear),
+      "close" => Ok(Command::Close),
+      "open" => Ok(Command::Open),
+      "status" => Ok(Command::Status),
+      _ => Err(()),
+    }
+  }
+}
+
+/// Executes the given parser on the given arguments and handles the result.
+///
+/// This macro executed the given argument parser on the given arguments using the standard output
+/// and error.  It then handles three cases:  If the result is `Ok`, it does nothing as the
+/// arguments have been parsed successfully.  If the result is `Err(0)`, it returns `Ok(())` as the
+/// help message has been printed by `argparse`.  Otherwise it returns `Err(Error::ArgparseError)`
+/// as `argparse` failed.
+macro_rules! parse_args {
+  ( $p:expr, $a:expr ) => {
+    if let Err(err) = $p.parse($a, &mut io::stdout(), &mut io::stderr()) {
+      match err {
+        0 => return Ok(()),
+        _ => return Err(Error::ArgparseError),
+      }
+    }
+  };
+}
+
+/// Inquire the status of the nitrokey.
+fn status(args: Vec<String>) -> Result<()> {
+  let mut parser = argparse::ArgumentParser::new();
+  parser.set_description("Print the status of the connected Nitrokey Storage");
+  parse_args!(parser, args);
+
+  commands::status()
+}
+
+/// Open the encrypted volume on the nitrokey.
+fn open(args: Vec<String>) -> Result<()> {
+  let mut parser = argparse::ArgumentParser::new();
+  parser.set_description("Opens the encrypted volume on a Nitrokey Storage");
+  parse_args!(parser, args);
+
+  commands::open()
+}
+
+/// Close the previously opened encrypted volume.
+fn close(args: Vec<String>) -> Result<()> {
+  let mut parser = argparse::ArgumentParser::new();
+  parser.set_description("Closes the encrypted volume on a Nitrokey Storage");
+  parse_args!(parser, args);
+
+  commands::close()
+}
+
+/// Clear the PIN stored when opening the nitrokey's encrypted volume.
+fn clear(args: Vec<String>) -> Result<()> {
+  let mut parser = argparse::ArgumentParser::new();
+  parser.set_description("Clears the cached passphrase");
+  parse_args!(parser, args);
+
+  commands::clear()
+}
+
+/// Parse the command-line arguments and return the selected command and the remaining arguments
+/// for the command.
+pub fn parse_arguments() -> (Command, Vec<String>) {
+  let mut command = Command::Status;
+  let mut args = vec![];
+  let mut parser = argparse::ArgumentParser::new();
+  parser.set_description("Provides access to a Nitrokey device");
+  let _ = parser.refer(&mut command).required().add_argument(
+    "command",
+    argparse::Store,
+    "the command to execute (clear|close|open|status)",
+  );
+  let _ = parser.refer(&mut args).add_argument(
+    "arguments",
+    argparse::List,
+    "the arguments for the command",
+  );
+  parser.stop_on_first_argument(true);
+  parser.parse_args_or_exit();
```

Comment:
I'd rather not have anything exit the program in some side module. Similarly, I think we should pass in the argument vector to this function such that environment accesses are all located in the main file and dependencies are explicit.